src/features/currentTodo.ts

```diff
@@ -1,10 +1,39 @@
-import { createSlice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
-const initialState = null as Todo | null;
+// we use string literal as a type to avoid mistype in future
+type RemoveTodoAction = { type: 'currentTodo/REMOVE' };
 
-export const currentTodoSlice = createSlice({
-  name: 'currentTodo',
-  initialState,
-  reducers: {},
+// payload is a typical name for an action data
+type SetTodoAction = {
+  type: 'currentTodo/SET';
+  payload: Todo;
+};
+
+// Action creator return type protect us from a mistype
+const removeTodo = (): RemoveTodoAction => ({ type: 'currentTodo/REMOVE' });
+
+const setTodo = (todo: Todo): SetTodoAction => ({
+  type: 'currentTodo/SET',
+  payload: todo,
 });
+
+// These actions will be used in the application
+export const actions = { setTodo, removeTodo };
+
+type State = Todo | null;
+type Action = SetTodoAction | RemoveTodoAction;
+
+const currentTodoReducer = (action: Action, state: State = null): State => {
```

Comment:
The `currentTodoReducer` function has its parameters in the wrong order. According to Redux convention, the first parameter should be `state` and the second should be `action`. The correct signature should be `const currentTodoReducer = (state: State = null, action: Action): State => {`.